### DISCUSSION TITLE
pullrequestするまでの手順について

### DISCUSSION THREAD

**Discussion by mogy-h** [if-mentor/next_fire_todo_5#60] [Q&A]
issue#25 サインアップUIの作成が終わったのですが、mainブランチのまま開発を進めてしまいました。
その後、開発用ブランチの作成ができずアドバイスいただきたいです。

あと、プルリクまでの流れが不安なので質問させてください。

開発ブランチ作成→開発後そのブランチにコミット→`git push origin master`を実行→プルリクエスト
でよかったでしょうか。

## 試したこと
 * `git checkout #25-create-signup-ui`を実行しても`Your branch is up to date with 'origin/main'.`
 と出てしまい、移動できない。

**Comment by naokitime** [comment]
## gitのスタッシュを使って解決する方法

gitでstashという機能があり、一時的に編集した内容を保管することができます。

①mainブランチで変更したコードをstashを使い保管庫へ移動させる。
②mainブランチへの変更内容がなくなるため、ブランチの切り替えができる。
③新しいブランチを作成する。（#○○_〜〜）
④保管庫へ移動させた変更したコードを新しいブランチに適応させる。

上記の方法で解決できないでしょうか？
もし、作成したコードが消えてしまうと怖いので、どこかに保存してもらって実装してもらえると嬉しいです。

## 参考URL
https://qiita.com/chihiro/items/f373873d5c2dfbd03250
https://www.sejuku.net/blog/71428

**Reply by mogy-h** [reply]
stashしてmainブランチを編集前に戻すことはできたのですが、そこからブランチの作成、移動ができなかったです。

## 試したこと
* stash後`git branch`を実行するとmain`とだけ出る

* `checkout -b`だと`error: switch ’b’ requires a value`と出る。

git難しい...
